TestFoundation/TestNSFileManager.swift

```diff
@@ -136,6 +137,38 @@ class TestNSFileManger : XCTestCase {
         }
     }
     
+    func test_pathEnumerator() {
+        let fm = NSFileManager.defaultManager()
+        let basePath = "/tmp/testdir"
+        let itemPath = "/tmp/testdir/item"
+        let basePath2 = "/tmp/testdir/path2"
+        let itemPath2 = "/tmp/testdir/path2/item"
+        
+        ignoreError { try fm.removeItemAtPath(basePath) }
+        
+        do {
+            try fm.createDirectoryAtPath(basePath, withIntermediateDirectories: false, attributes: nil)
+            try fm.createDirectoryAtPath(basePath2, withIntermediateDirectories: false, attributes: nil)
+
+            fm.createFileAtPath(itemPath, contents: NSData(), attributes: nil)
+            fm.createFileAtPath(itemPath2, contents: NSData(), attributes: nil)
+
+        } catch _ {
+            XCTFail()
+        }
+        
+        if let e = NSFileManager.defaultManager().enumeratorAtPath(basePath) {
+            var foundItems : [String] = []
+            while let item = e.nextObject() as? NSString {
+                    foundItems.append(item.bridge())
+            }
+            XCTAssertEqual(foundItems, ["item","path2","path2/item"])
```

Comment:
This expects the files to be sorted in a specific order which is OS dependent. Ubuntu does not give a promise that the files on disk are in the same order as Darwin; so the foundItems should be an un-ordered collection (ala Set).